### ISSUE TITLE
[Material] Picker does not use Title as Placeholder / Hint Text

### ISSUE THREAD

**Issue by Kornelis** [xamarin/Xamarin.Forms#5497]
### Description
The Little Things Playground Sample show the new Material Picker based on the Visual introduced in XF 3.6. But the Title Propery of the Picker is not used as Hint/Placeholder for the Material Picker.

### Steps to Reproduce

1. Checkout the LittleThingsPlay Sample 
2. Compile and Run the Sample
3. Navigate to the Pickers Page

### Expected Behavior
The Title set for "Monkey Picker" should be shown as Placeholder for the Picker as long as no item is select. After selection an Item, the Placeholder should be shown as Header

### Actual Behavior
This Picker is just empty and after selecting an Item it just shows the Item Text and no header.

### Basic Information

- Version with issue: Xamarin Forms 3.6.0.220655
- IDE: Visual Studio 2017 (15.9.8)
- Platform Target Frameworks: 
  - Android: API Level 28 (Android 9.0)

### Screenshots

![screenshot_1552043782](https://user-images.githubusercontent.com/683359/54027580-636b6200-41a2-11e9-8cf0-fa68c5457dac.png)
![screenshot_1552046512](https://user-images.githubusercontent.com/683359/54027581-636b6200-41a2-11e9-9a1a-057c7a2eb20d.png)
![screenshot_1552046515](https://user-images.githubusercontent.com/683359/54027582-636b6200-41a2-11e9-9776-fa3c77e64802.png)

### Reproduction Link

[Little Things Playground](https://github.com/davidortinau/TheLittleThingsPlayground/)

**Comment by PureWeen** [comment]
A little background on this one

We're still figuring out the best api for placeholder/title/selected elements with material

The thinking for material is to instead of using title we add specific material placeholder fields that are more consistent with material entry. 

```xaml
<TimePicker m:PlaceHolder="place holder" />
```

Instead of having title which does two things on android and only one thing on iOS. Also we're thinking of adding a nullable attached property for the selected time/date so that it's not just defaulted at midnight

**Comment by Kornelis** [comment]
It would be nice to have both, a Title and aPlaceholder (Hint). So in case no entry is selected it would show a hint like "Please select a Monkey..." and if a Monkey is selected (either by user or preselected via code) it would just show the Title "Monkey" like on Entry Elements

**Comment by PureWeen** [comment]
So you're thinking once they select an actual monkey the placeholder inside the entry just becomes a title?

Our plan is to expose all the different recommended labels for material

https://material.io/design/components/text-fields.html

![image](https://user-images.githubusercontent.com/5375137/54620946-90ced000-4a2c-11e9-8db3-e9044f5330fd.png)


For your use case I would see the 
PlaceHolder ->"Monkey"
HelperText => "Please Select a Monkey"

Title is a bit unfortunate for iOS since the native pickers there are different so it's usefulness on iOS breaks down a bit.

**Comment by Kornelis** [comment]
As the Picker is not a Textfield i'd prefer to have a hint that the user has to make a choice instead of enter some text. Obviously this gets clear once the user clicks the picker but in my opinion this would make it more clear in a mixed UI with Entries & Pickers where the user has to enter text or can choose from elements from a Picker. But having the Placeholder as Title would look awkward if I just select an Item and above the selection it says "Select a Monkey"

Here is a short sketch on how I imagine the Picker should behave / look:

![expectation](https://user-images.githubusercontent.com/683359/54623052-0b1e4500-4a6b-11e9-952e-702c22910feb.jpg)

Where 
Placeholder => "Select a monkey"
Title (Label?) => "Monkey" (just like i'd expect to have the other pickers labeld with Date / Time)

Not sure if this goes 100% with the Material Design, but I think it would look nicer instead of having a "Select a Monkey" as helper text below the Picker when the rest of the UI are just Entries with no HelperText

**Comment by wesoos** [comment]
Any updates on this?

**Comment by Diraekt** [comment]
My solution for the moment is:

` public class CustomMaterialPickerRenderer : MaterialPickerRenderer, IMaterialEntryRenderer
    {
        string IMaterialEntryRenderer.Placeholder
        {
            get
            {
                Picker element = base.Element;
                var placeholder = element?.Title;
                return placeholder ?? (placeholder = string.Empty);
            }
        }
    }`

**Comment by wicket5602** [comment]
Any news on this? :)

**Comment by PureWeen** [comment]
@wicket5602 it's been popped off the backlog and put on the queue

**Comment by wicket5602** [comment]
@PureWeen Awesome :)

**Comment by Cklauber** [comment]
@Pure

> My solution for the moment is:
> 
> `
> public class CustomMaterialPickerRenderer : MaterialPickerRenderer, IMaterialEntryRenderer
> {
> 
> ```
>     string IMaterialEntryRenderer.Placeholder
>     {
>         get
>         {
>             Picker element = base.Element;
>             var placeholder = element?.Title;
>             return placeholder ?? (placeholder = string.Empty);
>         }
>     }
> }
> ```
> 
> `

Any way you could give us more information on how to do that?

**Comment by Goksly** [comment]
To add to this, as it is a bit of a jarring experience, the background colour should also default to the same as other inputs... So we need not know its a picker until we select it. Unless you want to put a little down arrow to the right? But no need for the whole background to be a different shade.

**Comment by joshykautz** [comment]
I'm still hoping this can get implemented so that we can more closely follow material design principles.

**Comment by PureWeen** [comment]
Fixed by 
https://github.com/xamarin/Xamarin.Forms/pull/7880

Thanks for  reminder on this issue @joshykautz !!

**Comment by joshykautz** [comment]
No problem - Thank you for all the hard work @PureWeen  - we really appreciate it!

**Comment by tbrabec** [comment]
Any example of how to use the placeholder text with DatePicker?

**Comment by PureWeen** [comment]
@tbrabec the pickers can't be set to nullable (which is unfortunate) so once this is merged in

https://github.com/xamarin/Xamarin.Forms/pull/6947

Than that is a possibility

**Comment by tbrabec** [comment]
@PureWeen Great, I hope it's merged soon. Thanks.

**Comment by JamesRoche** [comment]
Has this been included in the current stable release?

**Comment by PureWeen** [comment]
Should be

**Comment by JamesRoche** [comment]
I am seeing it picker but not date picker. Is this expected? I can set Title from Xaml and C sharp but nothing on the class i can see in DatePicker? Turbo annoying to have such a non uniform ui :(

**Comment by PureWeen** [comment]
@JamesRoche can you log an issue?

**Comment by rbonestell** [comment]
@JamesRoche @PureWeen I've submitted the DatePicker and TimePicker bug report: https://github.com/xamarin/Xamarin.Forms/issues/9674#issue-569171922

**Comment by sajanmittal** [comment]
Is it supported in xamarin 4.1??

**Comment by rajinpangkalpundai** [comment]
Any updates about this?

**Comment by TechGuyGeek** [comment]
This seems to work for me on IOS but not Android 
but using the same code 
PickerLanguages.Title = AppResources.SelectLanguage;
PickerTheme.Title = AppResources.SelectTheme;
![image](https://user-images.githubusercontent.com/15261696/87876625-415a3480-c9d1-11ea-8306-e8803d4bbac1.png)

**Comment by JamesRoche** [comment]
Just updated and i dont see this?
